shared/teams/join-team/join-from-invite.tsx

```diff
@@ -38,82 +47,116 @@ const JoinFromInvite = (props: Props) => {
   const waiting = rpcWaiting && clickedJoin
   const wasWaiting = Container.usePrevious(waiting)
   const showSuccess = wasWaiting && !waiting && !error
+  const teamname = (details?.teamName.parts || []).join('.')
 
-  const body = showSuccess ? (
-    <Kb.Box2
-      direction="vertical"
-      style={styles.center}
-      fullWidth={true}
-      fullHeight={true}
-      gap="small"
-      centerChildren={true}
-    >
-      <Success teamname={teamname} />
-      <Kb.Button type="Dim" label="Close" onClick={onNavUp} style={styles.button} waiting={waiting} />
-    </Kb.Box2>
-  ) : (
-    <Kb.Box2
-      centerChildren={true}
-      direction="vertical"
-      fullHeight={true}
-      fullWidth={true}
-      gap="xtiny"
-      style={styles.body}
-    >
-      <Kb.Box style={styles.avatar}>
-        <Kb.Avatar
-          size={96}
-          teamname={teamname}
-          isTeam={true}
-          imageOverrideUrl={details.teamAvatars['square_192']}
-        />
-        {details.teamIsOpen && (
-          <Kb.Box2
-            direction="horizontal"
-            style={styles.meta}
-            fullWidth={!Styles.isMobile}
-            centerChildren={true}
-          >
-            <Kb.Meta backgroundColor={Styles.globalColors.green} title="open" size="Small" />
+  const body =
+    details === undefined ? (
+      loaded ? (
+        <Kb.Box2
```

Comment:
could you restructure this a little so that the box doesn't get repeated 5 times?